src/domain/eventSearch/utils.tsx

```diff
@@ -263,30 +311,44 @@ export const getEventSearchVariables = ({
     keywordAnd.push('yso:p4354');
   }
 
-  const keywordsParamName = MAPPED_KEYWORD_TERMS[searchSource];
-  const mappedCategories: string[] = categories
-    .map((category) => MAPPED_CATEGORIES[searchSource][category])
-    .filter((e) => e);
-  const mappedHobbyTypes: string[] =
-    hobbyTypes
-      ?.map((hobbyType) => MAPPED_COURSE_HOBBY_TYPES[hobbyType])
-      .filter((e) => e) || [];
+  const categoriesParamName = MAPPED_KEYWORD_TERMS[searchSource];
+
+  const categoryMap = MAPPED_CATEGORIES[searchSource];
+
+  const getMappedPropertyValues = (
+    list: string[],
+    map: Record<string, string>
+  ) =>
+    list?.reduce<string[]>(
+      (prev, val: string) => prev.concat(map[val]?.split(',') ?? []),
+      []
+    );
+
+  const mappedCategories = getMappedPropertyValues(categories, categoryMap);
+
+  const mappedHobbyTypes = getMappedPropertyValues(
+    hobbyTypes ?? [],
+    MAPPED_COURSE_HOBBY_TYPES
+  );
+
   const hasLocation = !isEmpty(divisions) || !isEmpty(places);
   const hasText = !isEmpty(text);
-  // Combine and add keywords
+
+  //free text search
+  const isEventsSearch = searchSource === EVENT_SEARCH_SOURCES.EVENTS;
+  const freeTextSearchParam =
+    isEventsSearch && hasLocation
+      ? { localOngoingAnd: text }
+      : { allOngoingAnd: text };
 
   return {
-    ...(hasText &&
-      (hasLocation ? { localOngoingAnd: text } : { allOngoingAnd: text })),
+    ...(hasText && freeTextSearchParam),
     ...(hasLocation && { division: divisions.sort() }),
     end,
     include,
     isFree: isFree || undefined,
-    [keywordsParamName]: [
-      ...(keyword ? keyword : []),
-      ...mappedCategories,
-      ...mappedHobbyTypes,
-    ],
+    [categoriesParamName]: [...(keyword ?? []), ...mappedCategories],
+    keywordOrSet3: [...mappedHobbyTypes],
```

Comment:
I don't think there is need to create new array and spread these values.

Reply:
fixed